pos_payment_terminal/static/src/js/pos_payment_terminal.js

```diff
@@ -113,6 +140,40 @@ odoo.define('pos_payment_terminal.pos_payment_terminal', function (require) {
             var vals = _orderproto.export_as_JSON.apply(this, arguments);
             vals['transactions'] = this.transactions || {};
             return vals;
+        },
+        get_paymentline: function(id){
+            var paymentlines = this.paymentlines.models;
+            for(var i = 0; i < paymentlines.length; i++){
+                if(paymentlines[i].cid === id){
+                    return paymentlines[i];
+                }
+            }
+            return null;
+        },
+    });
+    var _paymentlineproto = models.Paymentline.prototype;
+    models.Paymentline = models.Paymentline.extend({
+        initialize: function(attr, options){
+            this.terminal_transaction_id = false;
+            this.terminal_transaction_success = false;
+            this.terminal_transaction_status = false;
+            this.terminal_transaction_reference = false;
+            _paymentlineproto.initialize.apply(this, arguments);
+        },
+        init_from_JSON: function(json){
+            _paymentlineproto.init_from_JSON.apply(this, arguments);
+            this.transaction_id = json.transaction_id;
+            this.success = json.success;
+            this.status = json.status;
+            this.reference = json.reference;
+        },
+        export_as_JSON: function() {
+            var vals = _paymentlineproto.export_as_JSON.apply(this, arguments);
+            vals['transaction_id'] = this.transaction_id || false;
+            vals['success'] = this.success || false;
+            vals['status'] = this.status || false;
+            vals['reference'] = this.reference || false;
```

Comment:
`|| false` sounds suspicious. Because for sucess, we want to distinguish (null/undefined, false, true).

I'd say you can simply do 
`vals.transaction_id = this.transaction_id`